library/src/animationManager.cxx

```diff
@@ -185,6 +187,32 @@ void animationManager::Tick(double deltaTime)
   }
 }
 
+//----------------------------------------------------------------------------
+void animationManager::JumpToFrame(int frame, bool relative)
+{
+  if (relative)
+  {
+    this->CurrentTime += ((this->DeltaTime * this->SpeedFactor) * this->AnimationDirection) * frame;
+  }
+  else
+  {
+    switch (frame)
+    {
+      case -1:
+        this->CurrentTime = this->TimeRange[1];
+        break;
+      default:
+        this->CurrentTime = this->TimeRange[0] + (frame * this->DeltaTime);
+        break;
+    }
```

Comment:
Instead of special-casing `-1`, how about generalizing negative numbers going from the end so that `-2` is second-to-last and so on?

something like:
```suggestion
    if (frame < 0){
       frameCount = /* not sure the exact math */;
       frame += frameCount; // -1 -> n-1, -2 -> n-2, ...
    }
    this->CurrentTime = this->TimeRange[0] + (frame * this->DeltaTime);
```